File: 2022-2023/sourcecode/lab/lab7/GalleryAppJava/app/src/main/java/pl/udu/uwr/pum/galleryappjava/fragments/GalleryFragment.java
package pl.udu.uwr.pum.galleryappjava.fragments;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import pl.udu.uwr.pum.galleryappjava.R;
import pl.udu.uwr.pum.galleryappjava.adapter.GalleryAdapter;
import pl.udu.uwr.pum.galleryappjava.databinding.FragmentAddPictureBinding;
import pl.udu.uwr.pum.galleryappjava.databinding.FragmentGalleryBinding;
import pl.udu.uwr.pum.galleryappjava.db.DBHandler;


public class GalleryFragment extends Fragment {

    private FragmentGalleryBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentGalleryBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        DBHandler dbHandler = new DBHandler(requireContext());


        binding.recycler.setLayoutManager(new LinearLayoutManager(requireContext()));
        binding.recycler.setAdapter(new GalleryAdapter(dbHandler.getAllItems()));

    }
}

File: 2022-2023/sourcecode/lab/lab7/GalleryApp/app/src/main/java/pl/udu/uwr/pum/galleryapp/fragments/GalleryFragment.kt
package pl.udu.uwr.pum.galleryapp.fragments

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import pl.udu.uwr.pum.galleryapp.R
import pl.udu.uwr.pum.galleryapp.adapter.GalleryAdapter
import pl.udu.uwr.pum.galleryapp.dataProvider.DataProvider
import pl.udu.uwr.pum.galleryapp.databinding.FragmentAddPictureBinding
import pl.udu.uwr.pum.galleryapp.databinding.FragmentGalleryBinding

class GalleryFragment : Fragment() {

    private lateinit var binding: FragmentGalleryBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentGalleryBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.recycler.apply {
            layoutManager = LinearLayoutManager(requireContext())
            adapter = GalleryAdapter(DataProvider.dummyData)
        }
    }
}